src/main/java/com/productdock/library/gateway/book/BookService.java

```diff
@@ -13,29 +14,49 @@
 public record BookService(CatalogClient catalogClient, RentalClient rentalClient,
                           InventoryClient inventoryClient, BookDetailsResponseCombiner bookDetailsResponseCombiner) {
 
+    private static final String CLAIM_EMAIL = "email";
+
     @SneakyThrows
     public JsonNode getBookDetailsById(String bookId, String jwtToken) {
+        var jwt = JWTParser.parse(jwtToken);
+        var userId = jwt.getJWTClaimsSet().getClaim(CLAIM_EMAIL).toString();
```

Comment:
This code is repeated in the method below, meaning it should be extracted in a separate method.